apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -501,22 +501,48 @@ impl Display for LicenseEnforcementReport {
     }
 }
 
-/// License controls availability of certain features of the Router. It must be constructed from a base64 encoded JWT
+/// These are claims extracted from the License and represent ways Apollo limits the router's usage It must be constructed from a base64 encoded JWT
 /// This API experimental and is subject to change outside of semver.
 #[derive(Debug, Clone, Default)]
 pub struct License {
     pub(crate) claims: Option<Claims>,
 }
 
+/// Transactions Per Second limits. We talk as though this will be in seconds, but the Duration
+/// here is actually given to us in milliseconds via the License's JWT's claims
+#[derive(Builder, Copy, Clone, Debug, Eq, PartialEq)]
+pub(crate) struct TpsLimit {
+    capacity: usize,
+    duration: Duration,
+}
+
+// FIXME: better defaults set with some producty insight and move then to consts
+impl Default for TpsLimit {
```

Comment:
Does it make sense to have any defaults for this?

Reply:
I think so, but I could be wrong; this seems like a good way to set the product defaults to fall back on if they're not available in the jwt. What do you think? Maybe the LicenseLimits could make TpsLimits optional so that None is the default (which is the case for, say, free tier)

Reply:
updated to remove the Default and use Option<TpsLimits> instead